### ISSUE TITLE
UX: Loss of already entered data due to mouse movement

### ISSUE THREAD

**Issue by kaiwas** [FreeCAD/FreeCAD#13089]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

the counters in the sketcher have a problem that is present in all input fields in the program. With the value entered, if we move the mouse in the 3D window, the data is lost. Can this be somehow solved?
If the user has already entered data, he should not lose it due to accidental mouse movement.

You have to hold your breath and stop your heartbeat to prevent this from happening.

https://github.com/FreeCAD/FreeCAD/assets/32273833/8c30d31b-a114-48b9-948e-fb1f38d64348

### Full version info

```shell
0.22
```

### Subproject(s) affected?

Sketcher

### Anything else?

This problem occurs everywhere where a 3D window is involved.
Perhaps some kind of global setting is required that would fix the entered data (as soon as the user specified the numbers) and if the fixation is not required, the fixation could be canceled with the **Esс-key**

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Roy-043** [comment]
You need to press Tab or Enter to confirm your input.

**Comment by kaiwas** [comment]
I understand it.
But before pressing TAB, accidental mouse movement may occur.
There are times when you have to think about the entered value or whatever (it takes some time to enter).
Those. There may be a pause between entering the first character and the last one, as well as pressing the tab key itself.
Just experiment a few times without letting go of the mouse and you may realize that the problem may be quite serious. She's everywhere!

**Comment by mpfusion** [comment]
Another idea is to use some kind of hysteresis, so the user has to move the cursor at least by *x* amount away to trigger “freeing” the tool again and discarding the entered value.

**Comment by JohnOCFII** [comment]
I agree this is a frustrating issue.  I know of no other program that, after activating data entry into a text field, allows mouse movements to change that value.  Working with a laptop with a touch pad, it is VERY common to brush your hands over the touchpad, which changes values.  Super annoying.  I think there were some recent fixes done in the `Draft` workbench to address this in that WB, but this is a problem across many parts of FreeCAD.

**Comment by benj5378** [comment]
I might be able to look at this. @maxwxyz, can you assign me, such that I don't forget

**Comment by maxwxyz** [comment]
any news on this @benj5378 ?

**Comment by benj5378** [comment]
I had a brief look. The locking feature is already implemented for when entering has finished (after pressing tab for moving to next input).

It just need to be extended to also cover inputs currently being entered. I had a hard time finding the code location, and didn't get further
